lib/api/CDataFrameAnalyzer.cc

```diff
@@ -373,13 +384,38 @@ void CDataFrameAnalyzer::writeProgress(int progress,
     writer.flush();
 }
 
+void mapToVector(const TStrSizeUMap& map, TStrVec& vector) {
+    assert(vector.empty());
+    vector.resize(map.size());
+    for (const auto& entry : map) {
+        std::size_t index = entry.second;
+        if (index >= vector.size()) {
+            HANDLE_FATAL(<< "Index out of bounds: " << index);
+        }
+        vector[index] = entry.first;
+    }
+}
+
+void mapsToVectors(const TStrSizeUMapVec& maps, TStrVecVec& vectors) {
+    assert(vectors.empty());
+    vectors.resize(maps.size());
+    for (std::size_t i = 0; i < maps.size(); i++) {
```

Comment:
Done.

BTW, I thought this kind of optimization is done automatically by the compiler these days.
But of course I'll comply with the standards :)

Reply:
For size_t it makes no difference. However, for a user defined type the compiler is forced to call a different function which would usually create a temporary. Since it is just as easy to type `++i` as `i++` we just mandate that we always use `++i` if the return type isn't used to save having to think about it.